services/vision/vision.go

```diff
@@ -127,8 +128,19 @@ func FindFirstName(r robot.Robot) string {
 
 // FirstFromRobot returns the first vision service in this robot.
 func FirstFromRobot(r robot.Robot) (Service, error) {
-	name := FindFirstName(r)
```

Comment:
Echoing Cheuk, I think `FindFirstName` should be the function that is edited  rather than making the new `FirstFromRobotMain` function. Essentially just copy the loop over ResourceNames into `FindFirstName`, and then you won't have to change the body of `FirstFromRobot`.

The only things that use `FindFirstName` are this function, and also some tests, which also do not want to find the remote service name, but the main robot service name.

Reply:
I do think `FindFirstName` should still return a remote name if a local name cannot be found, otherwise the function should be `FindFirstLocalName`, and `FirstFromRobot` should still return a remote vision if a local one cannot be found, or we should name the function `FirstFromMainRobot` or `FirstFromLocalRobot`.

Reply:
I'm worried that'll bring the same problem around. Both (remote and local) default services would be `builtin` and if the user doesn't specifically change the name, then `FindFirstName` would just return the shortname "builtin". We'd be back to the 50/50 scenario when we call `FindRobot(r, "builtin")`

Reply:
Good point, Let's rename it then `FirstFromLocalRobot` to really specify that's what we're going for. Since the vision service is always default on every robot, I think the default expected behavior when calling `FirstFromRobot` is to get the local version, always. If they want the remote version, they can specify it with the general `FromRobot` function.

Reply:
thats sounds good to me